packages/block-library/src/image/edit.native.js

```diff
@@ -461,6 +463,31 @@ export class ImageEdit extends Component {
 		} );
 	}
 
+	onSelectURL( newURL ) {
+		const {
+			createErrorNotice,
+			imageDefaultSize,
+			setAttributes,
+		} = this.props;
+
+		// Use RN's Image.getSize to determine if URL is a valid image
+		RNImage.getSize(
+			newURL,
+			( width, height ) => {
+				setAttributes( {
+					url: newURL,
+					id: undefined,
+					width,
+					height,
+					sizeSlug: imageDefaultSize,
+				} );
+			},
+			() => {
+				createErrorNotice( __( 'Invalid URL. Image file not found.' ) );
```

Comment:
I think we could now change the error message since we're actually checking if the image exists by doing a request, wdyt?

```suggestion
				createErrorNotice( __( 'Image file not found.' ) );
```

Additionally, I think we could keep the previous code that was checking if the URL was valid via the `isURL` function. This way we could achieve two items:
1. Prevent requesting the image's size if the URL is not valid.
2. Display an accurate error message when the URL is not valid with:
```createErrorNotice( __( 'Invalid URL.' ) );```

Reply:
While testing I found out that it's possible to point to internal URLs when not providing a protocol in the URL (e.g. `AppIcon60x60@2x` when using WordPress-iOS app). Probably this is an edge case, but I'd rather prevent users to point to internal resources if possible.

https://user-images.githubusercontent.com/14905380/168306680-61d21146-f666-4450-bb7f-70e081fe5d93.mp4

Reply:
I confirm this issue was addressed in https://github.com/WordPress/gutenberg/pull/40334/commits/344c3409565bae2a52be6271d0d6008b40199c32 by checking if the value is an URL.